### DISCUSSION TITLE
IconLayer change atlas achieve gif effects

### DISCUSSION THREAD

**Discussion by HeiYanjing** [visgl/deck.gl#3514] [Q&A]
I have 5 pictures,these pictures can form  a gif picture 。

          new IconLayer({
            id:'city-layer',
            data
            pickable: true,
            iconAtlas:this.state.testIcon,
            iconMapping:UNIT_ICON_MAPPING,
            sizeScale: 150,
            getPosition: d => d.coordinates,
            getIcon: d => 'marker',
        }),

I set a Interval (100ms) and  in function ,every 200ms I will refresh the state to change the iconAtlas,and in console it mean successfully executed。
but In my map, it looks like a gif picture.

    _testIcon(){
        this.setState({
            testIcon:fIcon1,
        })
       var aaa =  window.setTimeout(()=>{
            this.setState({
                testIcon:fIcon2,
            })
        },200)
         var bbb= window.setTimeout(()=>{
            this.setState({
                testIcon:fIcon3,
            })
        },400)
        window.setTimeout(()=>{
            this.setState({
                testIcon:fIcon4,
            })
        },600),
        window.setTimeout(()=>{
            this.setState({
                testIcon:fIcon5,
            })
        },800)
      }

I feel this way is fool.
Does anyone have a better way to do it?

**Comment by grahambates-sp** [comment]
Could you do something like this (untested)?

```
let n = 0;
let clear = window.setInterval(() => {
    this.setState({testIcon :this.state.testIcon + 1})
    
    if (n)  > 5 window.clearInterval(clear);
})
```

**Comment by HeiYanjing** [comment]
> Could you do something like this (untested)?
> 
> ```
> let n = 0;
> let clear = window.setInterval(() => {
>     this.setState({testIcon :this.state.testIcon + 1})
>     
>     if (n)  > 5 window.clearInterval(clear);
> })
> ```

If add  time control
for example,change picture every 200ms
